5-design-patterns/homework/src/Client.ts

```diff
@@ -0,0 +1,71 @@
+import { ShipmentParams } from './interfaces';
+import { Shipment } from "./Shipment";
+import { Letter } from "./Letter";
+import { Package } from "./Package";
+import { Oversized } from "./Oversized";
+import { Shipper } from "./Shipper";
+import { AirEastShipper } from './AirEastShipper';
+import { ChicagoSprintShipper } from './ChicagoSprintShipper';
+import { PacificParcelShipper } from "./PacificParcelShipper";
+
+export class Client {
+    private readonly shipmentParams: ShipmentParams;
+    private ZIP_CODE_FIRST_NUMBERS = {
+        airEast: [1, 2, 3],
+        chicagoSprint: [4, 5, 6],
+        pacificParcel: [7, 8, 9]
+    }
+
+    constructor(shipmentParams: ShipmentParams) {
```

Comment:
I'd separate client/shipment creation, can you have a client which doesn't have shipment yet? Or its details? I suppose yes. Client isn't equal shipment.